src/mpid/ch4/shm/ipc/src/ipc_send.h

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright (C) by Argonne National Laboratory
+ *     See COPYRIGHT in top-level directory
+ */
+
+#ifndef IPC_SEND_H_INCLUDED
+#define IPC_SEND_H_INCLUDED
+
+#include "ch4_impl.h"
+#include "mpidimpl.h"
+#include "shm_control.h"
+#include "ipc_pre.h"
+#ifdef MPIDI_CH4_SHM_ENABLE_XPMEM
+#include "../xpmem/xpmem_send.h"
+#endif
+
+/*
+=== BEGIN_MPI_T_CVAR_INFO_BLOCK ===
+
+cvars:
+    - name        : MPIR_CVAR_CH4_XPMEM_LMT_MSG_SIZE
```

Comment:
XPMEM-specific CVAR?

Reply:
IPC forwards the send to XPMEM only when the message size is larger than `MPIR_CVAR_CH4_XPMEM_LMT_MSG_SIZE`. When the message is small, we might want to use POSIX or another submodule, however, the "another submodule"  is out of XPMEM's control. Thus, it is defined here. 

We might want to have some condition check API provided by each submodule instead of exposing the threshold. But it is unclear how to do it cleanly, and it might be irrelevant to the refactoring PR itself as the original code also defines it in SHM glue layer.

Reply:
The LMT size for XPMEM might be different from GPU. E.g., https://github.com/pmodels/mpich/pull/4540 defines one for GPU that is higher than XPMEM.

Reply:
I see.  This is fine, then.